libmachine/provision/suse.go

```diff
@@ -205,3 +182,34 @@ func (provisioner *SUSEProvisioner) Provision(swarmOptions swarm.Options, authOp
 	err = provisioner.Service("docker", serviceaction.Enable)
 	return err
 }
+
+func (provisioner *SUSEProvisioner) configureFirewall() error {
+	tcpPorts := "22 80 443 2376 2379 2380 6443 9099 9796 10250 10254 30000:32767"
+	udpPorts := "8472 30000:32767"
+	cmds := make([]string, 0)
+	if _, installed := provisioner.SSHCommand("rpm -q firewalld"); installed == nil {
+		tcpPorts := strings.ReplaceAll(tcpPorts, ":", "-")
+		udpPorts := strings.ReplaceAll(udpPorts, ":", "-")
+		for _, port := range strings.Split(tcpPorts, " ") {
+			cmds = append(cmds, fmt.Sprintf("sudo firewall-cmd --permanent --add-port=%s/tcp", port))
+		}
+		for _, port := range strings.Split(udpPorts, " ") {
+			cmds = append(cmds, fmt.Sprintf("sudo firewall-cmd --permanent --add-port=%s/udp", port))
+		}
+		cmds = append(cmds, "sudo firewall-cmd --reload")
+	} else {
+		cmds = append(cmds,
+			fmt.Sprintf(
+				`sudo sed -i 's/FW_SERVICES_EXT_TCP=".*"/FW_SERVICES_EXT_TCP="%s"/' /etc/sysconfig/SuSEfirewall2`,
+				tcpPorts),
+			fmt.Sprintf(
+				`sudo sed -i 's/FW_SERVICES_EXT_UDP=".*"/FW_SERVICES_EXT_UDP="%s"/' /etc/sysconfig/SuSEfirewall2`,
+				udpPorts))
+	}
```

Comment:
Thes requirements are mirrored in https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/node-requirements/#opening-suse-linux-ports. 

`FW_ROUTE=yes` is not required as enabling the option alone does nothing according to SUSE docs: https://github.com/openSUSE/susefirewall2/blob/8d916dd5a59031b978819e0443eec3918051a5a9/SuSEfirewall2.sysconfig#L128-L138. Since Rancher doesn't require `FW_MASQUERADE` or `FW_FORWARD` this seems like it can be safely ignored.

Restarting the firewall with `SuSEfirewall2` also does not seem to be required.

Reply:
```Restarting the firewall with SuSEfirewall2 also does not seem to be required.```

How does this get applied if we only modify `/etc/sysconfig/SuSEfirewall2`? According to https://en.opensuse.org/SuSEfirewall2 this still needs running `/sbin/SuSEfirewall2`. Apparently it works, but would be good to know why and how so we don't have to figure it out when debugging or on some other error

Reply:
After looking around online, I haven't been able to find a concrete answer for why this step wasn't required to get the provisioning working correctly.

I suspect it could be due to the fact that the `wicked` service is enabled by default, which may call `/etc/ppp/ipup` and take care of this restart automatically ([see page 20](http://keycorner.org/pub/text/doc/firewall2-ltr-0-9.pdf)), but as a precautionary measure I've pushed a new commit out that performs `sudo /sbin/SuSEfirewall2` after updating the firewall rules in this case.

Tested it on AWS / Azure and it provisions as expected!